pkg/api/types_test.go

```diff
@@ -198,3 +364,33 @@ func TestJSONTags(t *testing.T) {
 		t.Errorf("mismatch in struct tags for %T: %s", o, err.Error())
 	}
 }
+
+func TestMarshallYaml(t *testing.T) {
+	populatedOc := OpenShiftManagedCluster{}
+	populate.Walk(&populatedOc)
+
+	b, err := yaml.Marshal(populatedOc)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	if !bytes.Equal(b, marshalledYaml) {
+		t.Errorf("yaml.Marshall returned unexpected result\n%s\n%s\n", string(b), marshalledYaml)
+	}
+}
+
+func TestUnmarshallYaml(t *testing.T) {
+	t.Skip("skipping this test until we are able to unmarshall x509.Certificate")
```

Comment:
@jim-minter This is the test which requires unmarshalling the internal api struct from a curated yaml. `api.Config` contains `x509.Certificate`s and that is what breaks the unmarshall. This test is currently skipped: